linker/shared/src/main/scala/org/scalajs/linker/frontend/BaseLinker.scala

```diff
@@ -36,8 +36,10 @@ final class BaseLinker(config: CommonPhaseConfig, checkIR: Boolean) {
   import BaseLinker._
 
   private val irLoader = new FileIRLoader
-  private val analyzer =
-    new Analyzer(config, initial = true, checkIR = checkIR, failOnError = true, irLoader)
+  private val analyzer = {
+    val checkIRFor = Some(CheckingPhase.Compiler).filter(_ => checkIR)
```

Comment:
```suggestion
    val checkIRFor = if (checkIR) Some(CheckingPhase.Compiler) else None
```

? even if a bit longer, IMO easier to grok.